db_benchmark_test.go

```diff
@@ -109,30 +187,156 @@ func setup(b *testing.B) *sql.DB {
+// table v2. In case of any error detected, benchmarks fail immediatelly.
+func insertIntoReportedV2(b *testing.B, connection *sql.DB, i int, report *string) {
+	// following columns needs to be updated with data:
+	// 1 | org_id            | integer                     | not null  |
+	// 2 | account_number    | integer                     | not null  |
+	// 3 | cluster           | character(36)               | not null  |
+	// 4 | notification_type | integer                     | not null  |
+	// 5 | state             | integer                     | not null  |
+	// 6 | report            | character varying           | not null  |
+	// 7 | updated_at        | timestamp without time zone | not null  |
+	// 8 | notified_at       | timestamp without time zone | not null  |
+	// 9 | error_log         | character varying           |           |
+	// 10| event_type_id     | integer                     |           |
+
+	orgID := i % 1000                  // limited number of org IDs
+	accountNumber := orgID + 1         // can be different than org ID
+	clusterName := uuid.New().String() // unique
+	notificationTypeID := 1            // instant report
+	stateID := 1 + i%4                 // just four states can be used
+	updatedAt := time.Now()            // don't have to be unique
+	notifiedAt := time.Now()           // don't have to be unique
+	errorLog := ""                     // usually empty
+	eventTypeID := i%2 + 1             // just two event type targets possible
+
+	_, err := connection.Exec(insertIntoReportedV2Statement, orgID,
+		accountNumber, clusterName, notificationTypeID, stateID,
+		report, updatedAt, notifiedAt, errorLog, eventTypeID)
+
+	// check for any error, possible to exit immediatelly
+	if err != nil {
+		b.Fatal(err)
+	}
+}
+
+// runBenchmarkInsertIntoReportedTableV1 function perform several inserts into
+// reported table v1. In case of any error detected, benchmarks fail
+// immediatelly.
+func runBenchmarkInsertIntoReportedTableV1(b *testing.B, scaleFactor int, report *string) {
+	// retrieve DB connection
 	connection := setup(b)
 	if connection == nil {
 		b.Fatal()
 	}
 
+	// make sure we start with no DB table
+	execSQLStatement(b, connection, dropTableReportedV1)
+
+	// create reported table from scratch
+	execSQLStatement(b, connection, createTableReportedV1)
+
+	// good citizens cleanup properly
+	//defer execSQLStatement(b, connection, dropTableReportedV1)
+
+	// time to start benchmark
 	b.ResetTimer()
 
 	// perform DB benchmark
 	for i := 0; i < b.N; i++ {
+		for j := 0; j < scaleFactor; j++ {
+			insertIntoReportedV1(b, connection, i, report)
+		}
 	}
 }
 
-// BenchmarkBar is dummy ATM
-func BenchmarkBar(b *testing.B) {
+// runBenchmarkInsertIntoReportedTableV2 function perform several inserts into
+// reported table v2. In case of any error detected, benchmarks fail
+// immediatelly.
+func runBenchmarkInsertIntoReportedTableV2(b *testing.B, scaleFactor int, report *string) {
```

Comment:
what about having only on `runBenchmarkInsertIntoReportedTable` function and passing the `insertIntoReported` function as an argument too?

Reply:
good advice! will do it

Reply:
implemented